rand_core/src/block.rs

```diff
@@ -183,7 +183,8 @@ where <R as BlockRngCore>::Results: AsRef<[u32]> + AsMut<[u32]>
         let read_u64 = |results: &[u32], index| {
             if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
                 // requires little-endian CPU supporting unaligned reads:
-                unsafe { *(&results[index] as *const u32 as *const u64) }
+                let ptr: *const u64 = results[index..index+1].as_ptr() as *const u64;
```

Comment:
This range still only covers 1 element, not 2 elements, as it uses `..` instead of `..=`.

Reply:
Yeah, I fixed that in https://github.com/rust-random/rand/pull/784

Reply:
Didn't notice that commit :)